Descent3/game.cpp

```diff
@@ -1008,10 +1011,16 @@ void SetScreenMode(int sm, bool force_res_change) {
         scr_height = 480;
       }
 
-      if (rend_width != scr_width || rend_height != scr_height) {
+	  if (rend_width != scr_width || rend_height != scr_height ||
+          Game_window_res_width != Render_preferred_state.window_width ||
+          Game_window_res_height != Render_preferred_state.window_height ||
+          Game_fullscreen != Render_preferred_state.fullscreen) {
```

Comment:
This complex boolean should go into a dedicated function

Reply:
why, is that condition used multiple times?

Reply:
it doesn't matter. It would make more explicit in which case this condition is triggered